gluonnlp/model/attention_cell.py

```diff
@@ -456,6 +456,11 @@ def _compute_weight(self, F, query, key, mask=None):
             query = self._proj_query(query)
             if not self._luong_style:
                 key = self._proj_key(key)
+            else:
+                assert query.shape[-1] == key.shape[-1], 'Luong style attention requires key to ' \
```

Comment:
query.shape is only available if query is ndarray. you can do the check by checking whether `F == nd`.

Reply:
added.